### PR TITLE
added masking layer

### PR DESCRIPTION
**Pull request by Gurpreetsingh9465**
this is a PR regarding issue tensorflow/tfjs#1502 usage
`
var a = tf.tensor([[[1, 2, 3], [1, 2, 3], [1, 2, 3]], [[8, 8, 8], [1, 5, 7], [8, 8, 9]], [[8, 8, 8], [8, 8, 8], [8, 8, 8]]]);`
`
var test2 = tf.layers.masking({maskValue: 8}).apply(a);
`
`test2.print();
`

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-layers/525)

### CODE REVIEW

**Comment by Gurpreetsingh9465** [comment]
is this a bug?
`import * as tf from '@tensorflow/tfjs';`
`const model = tf.sequential();`
`let t = tf.tensor([`
`    [[1, 0, 0], [1, 1, 0], [1, 1, 1]],`
`    [[0, 0, 0], [1, 0, 0], [1, 1, 0]]]);`
`model.add(tf.layers.reLU({batchInputShape: t.shape}));`
`model.add(tf.layers.lstm({units: 4}));`
`let ans = model.predict(t);`
`ans.print();`

Error: 
`Property 'print' does not exist on type 'Tensor<Rank>[]'.`

**Comment by caisq** [comment]
@Gurpreetsingh9465 This seems to be due to the union type of the return value of `predict()`. You should do.

```typescript
const ans = model.predict(t) as Tensor;
ans.print();
```

**Comment by caisq** [comment]
@Gurpreetsingh9465 I see you've been making changes to address my comments. Please let me know when the PR is ready for me to review again. Also don't hesitate to ask if you need additional help.

**Comment by Gurpreetsingh9465** [comment]
Error by using
`let ans =model.predict(t) as Tensor;`
`Argument of type 'Tensor<Rank>' is not assignable to parameter of type 'Tensor<Rank> | Tensor<Rank>[]'.`

also tried
`let ans =model.predict(t as Tensor) as Tensor;`
`Argument of type 'Tensor<Rank>' is not assignable to parameter of type 'Tensor<Rank> | Tensor<Rank>[]'.`

**Comment by Gurpreetsingh9465** [comment]
thank you @caisq :smiley:
